app/validators/answers/date_answer_validator.py

```diff
@@ -0,0 +1,60 @@
+import re
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+
+from app import error_messages
+from app.validators.answers.answer_validator import AnswerValidator
+
+
+class DateAnswerValidator(AnswerValidator):
+    def validate(self):
+        super(DateAnswerValidator, self).validate()
+
+        if not self.is_offset_date_valid():
+            self.add_error(error_messages.INVALID_OFFSET_DATE)
+
+    def is_offset_date_valid(self):
+        if (
+            self.answer["type"] == "Date"
```

Comment:
`self.answer["type"] == "Date"` shouldn't be needed since you know it is Date by this point.